Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -3180,6 +3180,13 @@ void WebPageProxy::handleMouseEvent(const NativeWebMouseEvent& event)
         processNextQueuedMouseEvent();
 }
 
+void WebPageProxy::shouldFireMouseDidMoveOverElement(const NativeWebMouseEvent& event)
+{
+    sendWithAsyncReply(Messages::WebPage::PerformHitTestAndGetUserData { event }, [this, protectedThis = Ref { *this }](WebHitTestResultData&& hitTestResult, OptionSet<WebEventModifier>&& modifiers, UserData&& userData) {
+        this->mouseDidMoveOverElement(WTFMove(hitTestResult), WTFMove(modifiers).toRaw(), WTFMove(userData));
```

Comment:
Hm…I don't think it's technically correct to call `.toRaw()` after moving the modifiers, though I think in this case it shouldn't make a difference. Might be cleaner to simply pass `OptionSet<WebEventModifier>`, since it's just a byte anyways.

```suggestion
        this->mouseDidMoveOverElement(WTFMove(hitTestResult), modifiers.toRaw(), WTFMove(userData));
```

Reply:
Yes you're right, forgot to remove the move when I realized `WebPageProxy:: mouseDidMoveOverElement` expects an opaque modifier flag.

Reply:
I'll just pass the `OptionSet` by value instead

Reply:
Not sure if this is what you meant but, can we also change the `uint32_t` to an `OptionSet` in the `messages.in` file, if it doesn't increase the patch size too much? Otherwise, we should probably do it in a follow-up.

`MouseDidMoveOverElement(struct WebKit::WebHitTestResultData hitTestResultData, uint32_t modifiers, WebKit::UserData userData)` in `WebPageProxy.messages.in`.

Reply:
> Not sure if this is what you meant, but can we also change the uint32_t to an OptionSet in the messages.in file, if it doesn't increase the patch size too much? Otherwise, we should probably do it in a follow-up.

Yes this is what I meant. Let me look into this idea. I know at some point in the roundtrip from the web process the callstack starts speaking in terms of `uint32_t` modifierFlags instead of `OptionSet<>` but I'm not sure why... hopefully is a simple fix.

Reply:
Ah, so `EventHandler` calls into `WebCore::Chrome::mouseDidMoveOverElement`, which is why we can't `OptionSet<WebEventModifier>` everything.

We could adopt `OptionSet<PlatformEvent::Modifier>` in liue, but that's out of scope of this PR so I'll just add a TODO.

Reply:
Changed to pass the `OptionSet` by value in https://github.com/WebKit/WebKit/commit/b7607875e50a6089c8262d0f679251a1a5216add